main.js

```diff
@@ -1,60 +1,59 @@
-  {text: 'I want all of you to enjoy your cake, so...enjoy.', movie: 'The Godfather'},
-  {text: 'Frankly my dear, I don\’t give a damn.', movie: 'Gone with the Wind'},
-  {text: 'You don\’t understand! I coulda had class. I coulda been a contender. I could\’ve been somebody, instead of a bum, which is what I am.', movie: 'On the Waterfront'},
-  {text: 'Toto, I\’ve got a feeling we\’re not in Kansas anymore.', movie: 'The Wizard of Oz'},
-  {text: 'Here\’s looking at you, kid.', movie: 'Casablanca'},
-  {text: 'As God is my witness, I\’ll never be hungry again.', movie: 'Gone with the Wind'},
-  {text: 'Say \‘hello\’ to my little friend!', movie: 'Scarface'},
-  {text: 'I feel the need — the need for speed!', movie: 'Top Gun'},
-  {text: 'Houston, we have a problem.', movie: 'Apollo 13'},
-  {text: 'Nobody puts Baby in a corner.', movie: 'Dirty Dancing'},
-  {text: 'I\’ll get you, my pretty, and your little dog, too!', movie: 'The Wizard of Oz'},
-  {text: 'One morning I shot an elephant in my pajamas. How he got in my pajamas, I don\’t know.', movie: 'Animal Crackers'},
-  {text: 'You\’ve got to ask yourself one question: \‘Do I feel lucky?\’ Well, do ya, punk?', movie: 'Dirty Harry'},
-  {text: 'Hold on to your butts.', movie: 'Jurassic Park'},
-  {text: 'As far back as I can remember, I always wanted to be a gangster.', movie: 'Goodfellas'},
-  {text: 'Human sacrifice! Dogs and cats living together. Mass hysteria!', movie: 'Ghostbusters'}
-];
-const displayArea = document.getElementById('display-area');
-
-bindEvents();
-displayNewQuote();
-
-function bindEvents() {
-  const newQuoteButton = document.getElementById('new-quote-button');
-  newQuoteButton.addEventListener('click', handleNewQuoteClick);
-}
-
-function displayNewQuote() {
-  const quotePara = document.getElementById('quote');
-  const refCite = document.getElementById('ref');
-  const quote = getRandomQuote(quotes);
-
-  quotePara.innerText = '\“' + quote['text'] + '\”';
-  refCite.innerText = quote['movie'];
-  displayArea.classList.toggle('show');
-  updateTwitterIntent(quote);
-};
-
-function getRandomIndex(array) {
-  return Math.floor(Math.random() * array.length);
-}
-
-function getRandomQuote(quotes) {
-  const index = getRandomIndex(quotes);
-  return quotes[index];
-}
-
-function handleNewQuoteClick() {
-  displayArea.classList.toggle('show');
-  setTimeout(displayNewQuote, 350);
-}
-
-function updateTwitterIntent(quote) {
-  const twitterLink = document.getElementById('tweet-quote-link');
-  const intentAddress = `https://twitter.com/intent/tweet?text=“${quote.text}”  — ${quote.movie}`;
-  twitterLink.setAttribute('href', intentAddress);
+fetch('quotes.json')
+  .then(response => {
+    if (!response.ok) {
+      document.getElementById('quote-box').textContent = `
+        Network request failed with status 
+        ${response.status}: ${response.statusText}. 
+        Please refresh to try again.
```

Comment:
you chose to use line breaks here. What is your indentation rule? What do you think when do line breaks contribute to the reading flow? when do you think they disrupt the flow?

Reply:
I'm not sure what you mean by "indentation rule". Could you explain?

The main reason I used line breaks was because the template literal was running off the page, but I guess you could just turn on word wrap to fix that. I think the entire `fetch().then().then()` chain is cumbersome to read, but it's definitely easier without the line breaks in the template literal.

As to when line breaks contribute to/disrupt the reading flow: I would say it comes down to how much "cognitive load" a given line of code puts on the reader, how many statements the line contains and how easy/difficult each statement is to read.

Reply:
indentation is a [whole thing](https://en.wikipedia.org/wiki/Indentation_style). some companies have style guides which outline where a bracket goes, when to use line breaks, etc.. there are so called linters which do this job for you. here [an example](https://eslint.org). there is much discussion about these "coding styles". maybe you pick one for your own projects. personally, I do not care to much which coding style one uses as long as they are applied consistently/

Reply:
I agree with the disrupted reading flow. this is why I asked. maybe experiment a little with the code above. Everyone has different styles, but personally I value people who think about such details quite a bit more.

Reply:
there is an extension to javascript's `fetch.then.then`. it is called async/await. maybe do some research about it. I found [this one](https://wesbos.com/javascript/12-advanced-flow-control/70-async-await) right away. maybe check it out. please also check caniuse.com to be aware of how well supported the feature is.

Reply:
I am aware that my comments seem very detailed. however I think being able to discuss your coding style or relatively new technologies such as async/await may be deem very helpful in a work context later. (I sure would be impressed by a junior who is making an effort to maximise readability.